notebooks/registry.ipynb

```diff
@@ -0,0 +1,615 @@
+        "\n",
+        "The parameters passed in the command line will be parsed according to the type annotation of the corresponding low level constructor. If it is not one of `int`, `float`, `str`, or `bool`, it simply loads the string into python using a YAML loader.\n",
+        "\n",
+        "Each constructor returns the object, as well a dictionary config with all the parameters used to create the object and any `Registrable` objects created in the process of creating this object."
+      ],
+      "metadata": {
+        "id": "Cxo_vGT4-vJW"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Registering an Environment\n",
+        "\n",
+        "Consider registering a custom environment class named `Grid` (which inherits `BaseEnv`) in the RLHive registry. "
+      ],
+      "metadata": {
+        "id": "YrlRz676D_Dm"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from hive.envs.base import BaseEnv\n",
+        "class Grid(BaseEnv):\n",
+        "    def __init__(self, env_name = 'Grid', **kwargs):\n",
+        "        pass\n",
+        "    def reset(self):\n",
+        "        pass\n",
+        "    def step(self):\n",
+        "        pass\n",
+        "    def render(self):\n",
+        "        pass\n",
+        "    def close(self):\n",
+        "        pass\n",
+        "    def save(self):\n",
+        "        pass"
+      ],
+      "metadata": {
+        "id": "PbefnmFLJprJ"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from hive.utils.registry import registry\n",
+        "registry.register(name = 'Grid', \n",
+        "                  constructor = Grid, \n",
+        "                  type = BaseEnv)"
+      ],
+      "metadata": {
+        "id": "J3DGzZgP-u28"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "More than one environment can be registered at once using the `register_all` method. Consider registering three environments, `Gridv1`, `Gridv2`, and `Gridv3` in the RLHive registry."
+      ],
+      "metadata": {
+        "id": "sOy_VsfPLAIp"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "registry.register_all(\n",
+        "    BaseEnv,\n",
+        "    {\n",
+        "        \"Gridv1\": Gridv1,\n",
+        "        \"Gridv2\": Gridv2,\n",
+        "        \"Gridv3\": Gridv3,\n",
+        "    },\n",
+        ")"
+      ],
```

Comment:
This cell gives an error while executing "NameError: name 'Gridv1' is not defined", either define it or if you want to just give an example I think it would be better to put in a text cell. And the same goes for the following examples in the notebook.